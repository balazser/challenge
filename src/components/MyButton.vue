<script>
import Vue from 'vue'
import Component from 'vue-class-component'
import IconPaw from './../assets/paw.jpeg'

@Component({})
export default class MyButton extends Vue {
  handleClick() {
    this.$emit('click')
  }
  render() {
    return (
      <button
        onClick={() => this.handleClick()}
        style="border: 1px solid black; padding: 5px 15px; border-radius: 10px"
      >
        <img src={IconPaw} style="width: 30px; margin-right: 10px" />
        <span>Adopt</span>
      </button>
    )
  }
}
</script>
