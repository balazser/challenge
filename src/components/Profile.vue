<script>
import Vue from 'vue'
import Component from 'vue-class-component'
import MyButton from './MyButton.vue'
import paw from './../assets/paw.jpeg'

@Component({
  props: {
    id: String,
    name: String,
  },
})
export default class Profile extends Vue {
  hasAdopted = false

  handleClick() {
    this.hasAdopted = true
    this.$emit('adopted', this.id)
  }

  render() {
    return (
      <div>
        <img src={paw} style="margin-bottom: 15px" />
        <p>
          <strong>Name: </strong> {this.name}
        </p>
        <p>
          <strong>Has adopted: </strong> {this.hasAdopted ? 'yes' : 'no'}
        </p>
        <MyButton onClick={() => this.handleClick()} />
        <b-button>I am a Button</b-button>
      </div>
    )
  }
}
</script>
