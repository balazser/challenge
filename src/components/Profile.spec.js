import { mount, shallowMount } from '@vue/test-utils'
import Profile from './Profile.vue'
import MyButton from './MyButton.vue'

describe('Profile.vue', () => {
  it('should show the pet name', () => {
    const name = 'Doggy'
    const wrapper = shallowMount(Profile, {
      propsData: { name },
    })
    expect(wrapper.text()).toMatch(name)
  })
  it('should emit when pet has taken', () => {
    // arrange
    const idPet = '123'
    const wrapper = shallowMount(Profile, {
      propsData: {
        id: idPet,
      },
    })
    const button = wrapper.find(MyButton)

    // act
    button.trigger('click')

    // assert
    expect(wrapper.vm.hasAdopted).toBeTruthy()
    const emits = wrapper.emitted('adopted')
    expect(emits).toBeTruthy()
    const [[firstParameter]] = emits
    expect(firstParameter).toBe(idPet)
  })
  it('should match with snapshot', () => {
    const wrapper = mount(Profile)
    expect(wrapper.element).toMatchSnapshot()
  })
})
