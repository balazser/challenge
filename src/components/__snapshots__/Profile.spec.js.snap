// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue should match with snapshot 1`] = `
<div>
  <img
    src="./some_path.jpeg"
    style="margin-bottom: 15px;"
  />
  <p>
    <strong>
      Name: 
    </strong>
     
  </p>
  <p>
    <strong>
      Has adopted: 
    </strong>
     no
  </p>
  <button
    style="border: 1px solid black; padding: 5px 15px; border-radius: 10px;"
  >
    <img
      src="./some_path.jpeg"
      style="width: 30px; margin-right: 10px;"
    />
    <span>
      Adopt
    </span>
  </button>
</div>
`;
